packages/next/src/server/load-components.ts

```diff
@@ -108,14 +108,18 @@ async function loadComponentsImpl<N = any>({
   page: string
   isAppPath: boolean
 }): Promise<LoadComponentsReturnType<N>> {
-  let DocumentMod = {}
-  let AppMod = {}
+  let Document: DocumentType | undefined
+  let App: AppType | undefined
   if (!isAppPath) {
-    ;[DocumentMod, AppMod] = await Promise.all([
+    const [DocumentMod, AppMod] = await Promise.all([
       Promise.resolve().then(() => requirePage('/_document', distDir, false)),
       Promise.resolve().then(() => requirePage('/_app', distDir, false)),
     ])
+
+    Document = interopDefault(DocumentMod)
+    App = interopDefault(AppMod)
```

Comment:
These interoped values are now correctly including `undefined`, ensuring that downstream methods check the existence of the values prior to usage.